src/merkle_hash.rs

```diff
@@ -0,0 +1,2 @@
+/// The MerkleHash type is a string that must be a created by doing a SHA256 hash of an element.
+pub type MerkleHash = String;
```

Comment:
You could restrict the `MerkleHash` to be a 32 bit array. Otherwise, any `String` could be considered a valid digest.

Reply:
I thought about this while I was developing it, actually initialy I used to handle the `MerkleHash` type as an `[u8;32],` but I changed it because it's easier to parse the hashes that the user sends this way, I think that overall its better for the user experience. I do give the option to the user to pass a string and let the merkle tree do the hashing of it before adding it to the stree, with `add <element> <--hash>`. What do you think of it?

Reply:
I think it’s good for the user to have that flexibility. However, when creating a Merkle Tree from a file with a newline at the end, the newline is recognized as an element of the tree. It would be better to filter out blank lines so they aren’t added as elements accidentally.

Reply:
Done!